### ISSUE TITLE
[Question] How to use one reference screenshot for multiple browsers for visual tests

### ISSUE THREAD

**Issue by Dinerville** [microsoft/playwright#13812]
Hello

I'm creating visual tests for components in my storybook and I'm using two browsers to run tests : webkit and chromium

Playwright creates 2 screenshots for each test: one for webkit and one for chromium.
In some cases, screenshots are identical, in others - not. 

I would like to store only one reference screenshot for those cases when screenshots are identical and 2 screenshots where they are not so that I can save my storage space as I have a few thousands of screenshots that occupy quite a lot of storage space

**I was looking for something like that**

- When they are not identical
```js
expect(await page.screenshot()).toMatchSnapshot({
 linux: {
   webkit : 'screenshot-for-webkit',
   chromium : 'screenshot-for-chromium'
 }
});
```
and files will look like `screenshot-for-webkit.png` and `screenshot-for-chromium.png`

- When they are identical:
```js
expect(await page.screenshot()).toMatchSnapshot({
 linux: {
   webkit : 'screenshot-for-webkit-and-chromium',
   chromium :'screenshot-for-webkit-and-chromium',
 } 
});
```
and the file is `screenshot-for-webkit-and-chromium.png`

Is it possible? Are there any other solutions?

**Comment by dgozman** [comment]
This is not possible yet, but we are exploring solutions for this and similar issues around screenshots.

**Comment by aslushnikov** [comment]
The https://github.com/microsoft/playwright/issues/7792 will be fixed with the upcoming v1.28 with the `snapshotPathTemplate` configuration option.

With this option, you'll be able to do the following:

```ts
// playwright.config.ts
export default {
  snapshotPathTemplate: `__screenshots__/{projectName}/{arg}{ext}`,
};
```

And in your tests:
```ts
// for identical screenshots:
await expect(page).toHaveScreenshot('../common/screenshot-for-webkit-and-chromium.png');
// for per-project screenshots:
await expect(page).toHaveScreenshot('screenshot.png');
```

So merging in https://github.com/microsoft/playwright/issues/7792
